pennylane/qnode_new/decorator.py

```diff
@@ -47,30 +57,41 @@ def QNode(
               TensorFlow ``tfe.Variable`` objects.
 
         mutable (bool): whether the QNode circuit is mutable
-        differentiable (bool): whether the QNode is differentiable
+        diff (str, None): the method of differentiation to use in the created QNode.
+
+            * ``"best"``: Best available method. Uses the device directly to compute
+              the gradient if supported, otherwise will use the analytic parameter-shift
+              rule where possible with finite-difference as a fallback.
+
+            * ``"parameter-shift"``: Use the analytic parameter-shift
+              rule where possible with finite-difference as a fallback.
+
+            * ``"finite-diff"``: Uses numerical finite-differences.
+
+            * ``None``: a non-differentiable QNode is returned.
+
         properties (dict[str->Any]): additional keyword properties passed to the QNode
     """
-    if not differentiable:
+    if diff is None:
         # QNode is not differentiable
         return BaseQNode(func, device, mutable=mutable, properties=properties)
 
-    # Query the device to determine what analytic Jacobian QNodes
-    # are supported.
-
-    # TODO: should the device declare supported QNode
-    # classes, as well as the default QNode to use?
-
     # Set the default model to qubit, for backwards compatability with existing plugins
     # TODO: once all plugins have been updated to add `model` to their
-    # capabilities plugin, change the logic here.
+    # capabilities dictionary, update the logic here
     model = device.capabilities().get("model", "qubit")
+    device_jacobian = device.capabilities().get("provides_jacobian", False)
+
+    if device_jacobian and (diff == "best"):

```

Comment:
The parentheses are not needed here

Reply:
Yep, that's true. I tend to parenthesize boolean expressions to make them easier to read.